parcels/particleset/particlesetnodes.py

```diff
@@ -0,0 +1,1389 @@
+                while add_iter < len(self.repeatlon):
+                    gen_id = None if self.repeatpid is None or type(self.repeatpid) not in [list, tuple, dict, np.ndarray] else self.repeatid[add_iter]
+                    lon = self.repeatlon[add_iter]
+                    lat = self.repeatlat[add_iter]
+                    pdepth = self.repeatdepth[add_iter]
+                    ptime = time if self.release_starttime is None else self.release_starttime
+                    pid = self._idgen.nextID(lon, lat, pdepth, ptime) if gen_id is None else gen_id
+                    pdata = self.repeatpclass(lon, lat, pid=pid, ngrids=ngrids, depth=pdepth, time=ptime)
+                    pdata.dt = dt
+                    # Set other Variables if provided
+                    for kwvar in self.repeatkwargs:
+                        if isinstance(kwvar, Field):
+                            continue
+                        if not hasattr(pdata, kwvar):
+                            raise RuntimeError('Particle class does not have Variable %s' % kwvar)
+                        setattr(pdata, kwvar, self.repeatkwargs[kwvar][add_iter])
+                    self.add(pdata)
+                    add_iter += 1
+                next_prelease += self.repeatdt * np.sign(dt)
+            if abs(time-next_output) < tol or execute_once:
+                for fld in self._fieldset.get_fields():
+                    if hasattr(fld, 'to_write') and fld.to_write:
+                        if fld.grid.tdim > 1:
+                            raise RuntimeError('Field writing during execution only works for Fields with one snapshot in time')
+                        fldfilename = str(output_file.name).replace('.nc', '_%.4d' % fld.to_write)  # what does this do ? the variable is boolean, then it's increased - what-the-frog ...
+                        fld.write(fldfilename)
+                        fld.to_write += 1
+                if output_file is not None:
+                    output_file.write(self, time)
+                next_output += outputdt * np.sign(dt)
+            if abs(time-next_movie) < tol:
+                self.show(field=movie_background_field, show_time=time, animation=True)
+                next_movie += moviedt * np.sign(dt)
+            # ==== insert post-process here to also allow for memory clean-up via external func ==== #
+            if abs(time-next_callback) < tol:
+                if postIterationCallbacks is not None:
+                    for extFunc in postIterationCallbacks:
+                        extFunc()
+                next_callback += callbackdt * np.sign(dt)
+            if time != endtime:
+                next_input = self.fieldset.computeTimeChunk(time, dt)
+            if dt == 0:
+                break
+            if verbose_progress:
+                pbar.update(abs(time - _starttime))
+        if output_file is not None:
+            output_file.write(self, time)
+        if verbose_progress:
+            pbar.finish()
```

Comment:
Probably need to implement the new `tqdm` progressbar call (`pbar.close()`) here and also on updating above, see #1111?